apps/meteor/server/methods/browseChannels.js

```diff
@@ -143,19 +149,23 @@ async function getTeams(user, searchTerm, sort, pagination) {
 		},
 	);
 
-	const [rooms, total] = await Promise.all([
-		cursor
-			.map((room) => ({
-				...room,
-				roomsCount: getChannelsCountForTeam(room.teamId),
-			}))
-			.toArray(),
-		totalCount,
-	]);
+	const roomsPromises = cursor.map((room) => room).toArray();
+
+	const [rooms] = await Promise.all([roomsPromises]);
+
+	const teams = [];
+
+	for await (const room of rooms) {
+		const result = await getChannelsCountForTeam(room.teamId).toArray();
+
+		teams.push({ ...room, roomsCount: result.length });
+	}
+
+	const [total] = await Promise.all([totalCount]);
```

Comment:
```suggestion
	const [rooms, total] = await Promise.all([cursor.toArray(), totalCount]);
	const teams = [];

	for (const room of rooms) {
		const roomsCount = await getChannelsCountForTeam(room.teamId).count();
		teams.push({ ...room, roomsCount });
	}
```
I think we can use `Promise.all` here to get all the data at once. Also, I think we don't need to use a `for await` (I've tested it and it looks like just a `for` should be enough here).